build_tools/doctool/src/com/google/doctool/custom/JavaEmulSummaryDoclet.java

```diff
@@ -0,0 +1,273 @@
+                        .replace(".", "/") + "/";
+
+                Iterator<? extends Element> classesIterator = pack.getEnclosedElements()
+                        .stream()
+                        .filter(element -> env.isSelected(element) && env.isIncluded(element))
+                        .filter(element -> element.getModifiers().contains(Modifier.PUBLIC))
+                        .sorted(Comparator.comparing((Element o) -> o.getSimpleName().toString()))
+                        .iterator();
+
+                while (classesIterator.hasNext()) {
+                    Element cls = classesIterator.next();
+                    // Each class links to Oracle's main JavaDoc
+                    emitClassDocs(env, pw, packURL, cls);
+                    if (classesIterator.hasNext()) {
+                        pw.print("\n");
+                    }
+                }
+
+                pw.println("</dl>\n");
+            });
+
+            pw.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    private void emitClassDocs(DocletEnvironment env, PrintWriter pw, String packURL, Element cls) {
+        pw.format("  <dt><a href=\"%s%s.html\">%s</a></dt>\n", packURL,
+                qualifiedSimpleName(cls), qualifiedSimpleName(cls));
+
+        // Print out all fields
+        String fields = cls.getEnclosedElements()
+                .stream()
+                .filter(element -> element.getKind().isField())
+                .filter(field -> field.getModifiers().contains(Modifier.PUBLIC))
+                .map(field -> field.getSimpleName().toString())
+                .collect(Collectors.joining(", "));
+
+        if (!fields.isEmpty()) {
+            pw.format("  <dd style='margin-bottom: 0.5em;'>%s</dd>\n", fields);
+        }
+
+        List<String> constructors = cls.getEnclosedElements()
+                .stream()
+                .filter(element -> ElementKind.CONSTRUCTOR == element.getKind())
+                .filter(member -> member.getModifiers().contains(Modifier.PUBLIC))
+                .map(member -> (ExecutableElement) member)
+                .map(executableElement -> flatSignature(env, cls, executableElement))
+                .collect(Collectors.toList());
+
+        List<String> methods = cls.getEnclosedElements()
+                .stream()
+                .filter(element -> ElementKind.METHOD == element.getKind())
+                .filter(member -> member.getModifiers().contains(Modifier.PUBLIC))
+                .map(member -> (ExecutableElement) member)
+                .map(executableElement -> flatSignature(env, cls, executableElement))
+                .collect(Collectors.toList());
+
+        List<String> members = new ArrayList<>(constructors);
+        members.addAll(methods);
+
+        // Print out all constructors and methods
+        if (!members.isEmpty()) {
+            pw.format("  <dd>%s</dd>\n", String.join(", ", members));
```

Comment:
If this was higher throughput, I'd strongly encourage writing each member rather than join and then write.